commands/args.go

```diff
@@ -171,7 +168,8 @@ func NewArgs(args []string) *Args {
 		globalFlags []string
 	)
 
-	slurpGlobalFlags(&args, &globalFlags, &noop)
+	slurpGlobalFlags(&args, &globalFlags)
+	noop = removeValue(&globalFlags, "--noop")
```

Comment:
I think `removeValue` could be in `slurpGlobalFlags`'s for loop as well by iterating with index only (no `range`):

```go
for i := len(*args) - 1; i >= 0; i-- {
  arg := *args[i] 
  // other ifs
  if arg == "--noop" {
    *args, _ = removeItem(*args, i) // note that we may not need removeValue since removeItem does the job here.
  }
}
```

What do you think?